R/doeAnalysis.R

```diff
@@ -550,6 +562,70 @@ doeAnalysis <- function(jaspResults, dataset, options, ...) {
   )
 }
 
+.doeAnalysisPlotContourSurface <- function(jaspResults, dataset, options, ready) {
+  if (!is.null(jaspResults[["contourSurfacePlot"]]) || !options[["contourSurfacePlot"]]) {
+    return()
+  }
+  plotType <- options[["contourSurfacePlotType"]]
+  plotTypeString <- ifelse(plotType == "contourPlot", gettext("Contour plot"), gettext("Surface plot"))
+  containerTitle <- ifelse(plotType == "contourPlot", gettext("Contour plots"), gettext("Surface plots"))
+  container <- createJaspContainer(title = containerTitle)
+  container$dependOn(options = c("contourSurfacePlot", "contourSurfacePlotType",
+                                                            "contourSurfacePlotVariables", "contourSurfacePlotLegend",
+                                                            "contourSurfacePlotResponseDivision", "surfacePlotVerticalRotation",
+                                                            "surfacePlotHorizontalRotation", .doeAnalysisBaseDependencies()))
+  container$position <- 12
+  jaspResults[["contourSurfacePlot"]] <- container
+  
+  if (!ready || is.null(jaspResults[["doeResult"]]) || jaspResults$getError() ||
+      length(options[["contourSurfacePlotVariables"]]) < 2) {
+    plot <- createJaspPlot(title = plotTypeString)
+    jaspResults[["contourSurfacePlot"]][["plot"]] <- plot
+    return()
+  }
+  
+  variables <- unlist(options[["contourSurfacePlotVariables"]])
+  variablePairs <- t(utils::combn(variables, 2))
```

Comment:
This will be very, very inefficient for many variables. The complexity, but also the number of output objects, is `2^length(variables)`. Is that expected and are you sure people won't try to put 7 variables in there?

Reply:
It can only be as many variables as they have continuous predictors in their model. I would be surprised if people regularly have very excessive amounts in there. An alternative would be to make the input field a pairwise field, so people have to specify the combinations they want to see by hand? But for the cases with 2-4 variables that is usually all I would think.